stor/dx.py

```diff
@@ -824,12 +828,60 @@ def upload(self, to_upload, **kwargs):
                 raise NotFoundError('Source path ({}) does not exist. Please provide '
                                     'a valid source'.format(upload_obj.source))
 
-    def open(self, mode='r', encoding=None):
+    def open(self, mode='r', encoding=None, **kwargs):
         """
-        Opens a OBSFile that can be read or written to and is uploaded to
-        the remote service.
+        Args:
+            mode (str): The mode of the resource. Can be "r" for
+                reading the resource, "w" for writing the
+                resource, and "a" for appending to the open resource.
+            encoding (str): the text encoding to use on read/write, defaults to
+                ``locale.getpreferredencoding(False)`` if not set. We *strongly* encourage you to
+                explicitly set an encoding when reading/writing text (because
+                writers from different computers may store data on DNAnexus in different ways).
+                Python 3 only.
+        """
+        if mode == 'a':
```

Comment:
append mode can't be supported for OBSPaths generally - would it make more sense to move open to the shared baseclass obs paths?